.buildkite/filter-pipeline.nix

```diff
@@ -0,0 +1,34 @@
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+{ git_diff }:
```

Comment:
You calculate diff with `origin/master` in the buidlkite step, but `origin/master` in buildkite's checkout may be stale because it only fetches the branch it needs, you probably need to fetch master explicitly

Reply:
hmm, apparently, fetching `origin/master` is a bad idea as well because we probably want to compare our changes with the base branch and not with the existing master, e.g. in case where a feature branch is outdated and needs to be rebased

Reply:
If you use `...` and not `..` in `git diff`, it will compare to the common base, where the branch forked off

Reply:
looks like the proper way is to fetch `origin/master` and do `git diff --name-only $(git merge-base --fork-point origin/master)`

Reply:
looks like I'm wrong because `git diff origin/master...HEAD` uses `git merge-base` inside